hybridse/src/udf/udf.cc

```diff
@@ -64,6 +64,45 @@ void hex(StringRef *str, StringRef *output) {
     output->data_ = buffer;
 }
 
+void unhex(StringRef *str, StringRef *output) {
+    std::ostringstream ss;
+    uint8_t* arr = new uint8_t[str->size_];
+    memset(arr, 0, str->size_);
+    bool flag = 0;
+    for (uint32_t i=0; i < str->size_; i++) {
+        // only recognize 'A' to 'F' and 'a' to 'f' and '0' to '9', otherwise return null.
+        if (str->data_[i] <= 'F' && str->data_[i] >= 'A') {
+            arr[i] = str->data_[i] - 'A' + 10;
+        } else if (str->data_[i] <= 'f' && str->data_[i] >= 'a') {
+            arr[i] = str->data_[i] - 'a' + 10;
+        } else if (str->data_[i] <= '9' && str->data_[i] >= '0') {
+            arr[i] = str->data_[i] - '0';
+        } else {
+            flag = 1;
+            break;
+        }
+    }
+    if (flag == 0) {    // every character is valid hex character
```

Comment:
here is the behavior of spark sql:
```
spark-sql> select unhex(667), char_length(unhex(667)), hex(unhex(667));
g	2	6700
Time taken: 0.049 seconds, Fetched 1 row(s)
```
strange behavior but it seem interrupt it as `67`, `00`. But I think our behavior is fine too

Reply:
also to metion, mysql has more similar behavior to us:
```
MariaDB [(none)]> select char_length(unhex(667)), unhex(667), hex(unhex(667));
+-------------------------+------------+-----------------+
| char_length(unhex(667)) | unhex(667) | hex(unhex(667)) |
+-------------------------+------------+-----------------+
|                       2 | g         | 0667            |
+-------------------------+------------+-----------------+
```